### ISSUE TITLE
Session time out

### ISSUE THREAD

**Issue by vicksvmw** [darkoperator/Posh-SSH#128]
I am trying to SSH into HPiLO through powershell.
I am using below commands and every time it sessions out. 

 New-SSHSession -ComputerName 10.42.187.xx -Verbose

Errors:
New-SSHSession : Session operation has timed out
At line:1 char:1
+ New-SSHSession -ComputerName 10.42.187.xx -Verbose
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationTimeout: (Renci.SshNet.SshClient:SshClient) [New-SSHSession], SshOperationTimeoutException
    + FullyQualifiedErrorId : SSH.NewSshSession

New-SSHSession : Socket read operation has timed out after 10000 milliseconds.
At line:1 char:1

Can you please help..

Bascially I am trying below:

New-SSHSession -ComputerName 10.42.187.xx -Verbose -Credential $cred

Invoke-SSHCommand -Command "cd /map1/firmware1" -SSHSession | load –source http://10.118.161.xx/ilo2_229.bin

**Comment by darkoperator** [comment]
I believe in this case it would be better a stream

**Comment by darkoperator** [comment]
are you sure that SSH is opened on the HPiLO? did you tried using Putty to validate?

**Comment by darthVikes** [comment]
So I have a case where I'm ssh'ing in and it works fine via Putty as it loads more interactively but the profile on the server side as I'm using OpenSSH on windows using PowerShell as a subsystem and because the profile is being loaded it takes a little longer to connect.   So I added -ConnectionTimeout 300 and -OperationTimeout 300 and it then connected as it waited longer so maybe try adding those options?

**Comment by darthVikes** [comment]
Right, that's what fixed it for me as it was taking longer to load / connect.
